src/creator.ts

```diff
@@ -238,6 +238,23 @@ export class SlashCreator extends (EventEmitter as any as new () => TypedEventEm
     this.emit('debug', 'Server started');
   }
 
+  /**
+   * Registers a global component callback. Note that this will have no expiration, and should be invoked by the returned name.
+   * @param custom_id The custom ID of the component to register
+   * @param callback The callback to use on interaction
+   */
+  registerGlobalComponent(custom_id: string, callback: ComponentRegisterCallback) {
+    if (this._componentCallbacks.has(custom_id))
+      throw new Error(`A global component with the ID "${custom_id}" is already registered.`);
+    const newName = `global-${custom_id}`;
+    this._componentCallbacks.set(newName, {
```

Comment:
If you're declaring a new identifier - shouldn't that be put above everything else to be called on by the precondition?
- L247 - checks with `custom_id`
  - L248 - throws if true with `custom_id` in the reason
- L249 - declare `newName` using `custom_id`
- L250 - register the callback with the internal map **using `newName`**

Reply:
I'm sorry, but I am unable to comprehend your suggestion or what it would change on the code mentioned. Would you mind demonstrating what you mean?